core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java

```diff
@@ -460,15 +460,19 @@ private QuarkusControllerConfiguration createControllerConfiguration(
                                 "namespaces", v -> new HashSet<>(
                                         Arrays.asList(v.asStringArray())),
                                 () -> namespaces);
-                        cfg = new QuarkusKubernetesDependentResourceConfig(dependentNamespaces.toArray(new String[0]),
-                                labelSelector);
+                        cfg = new QuarkusKubernetesDependentResourceConfig(dependentNamespaces, labelSelector);
                     }
 
                     var nameField = dependentConfig.value("name");
                     final var name = Optional.ofNullable(nameField)
                             .map(AnnotationValue::asString)
                             .filter(Predicate.not(String::isBlank))
                             .orElse(DependentResource.defaultNameFor(dependentClass));
+                    final var spec = dependentResources.get(name);
+                    if (spec != null) {
```

Comment:
Why this change is needed?

Reply:
It wasn't needed but it's similar to what the SDK does: it shouldn't be possible to add dependent resources with the same name several times.